### PR TITLE
S87.26 add dropbox

### PR DESCRIPTION
**Pull request by ronoudgenoeg**

### CODE REVIEW

File: Lib/Dropbox/DropboxAuthContainer.php [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+
+/**
+ * Class ExampleMediaPlatformAuthContainer
+ *
+ * This class will be returned to dependencies of the AuthManager plugin.
+ * The authenticated API object will be stored by this class and used by the dependencies to communicate
+ * with the Media Platform API.
+ * This class has no methods because every API is different.
+ * The only thing required to set is the Media Platform user ID.
+ */
```
Comment (swordbeta) [inline Lib/Dropbox/DropboxAuthContainer.php]: Fix docblock

File: Lib/Dropbox/DropboxAuthManager.php [hunk 2]
```diff
@@ -0,0 +1,91 @@
+<?php
+App::uses('MediaPlatformAuthManager', 'AuthManager.Lib');
+App::uses('DropboxOauthProvider', 'AuthManager.Lib/Dropbox');
+App::uses('DropboxAuthContainer', 'AuthManager.Lib/Dropbox');
+
+/**
+ * Class ExampleMediaPlatformAuthManager
+ */
```
Comment (swordbeta) [inline Lib/Dropbox/DropboxAuthManager.php]: Fix docblock

File: Lib/Dropbox/DropboxAuthManager.php [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+
+		/**
+		 * @var \League\OAuth2\Client\Token\AccessToken|boolean $tokens
+		 */
+		$tokens = $this->_getAccessToken($request);
+		if (!$tokens) {
+			return false;
+		}
+		$this->_dropboxProvider->authorizationHeader = 'Bearer';
+		$details = $this->_dropboxProvider->getUserDetails($tokens);;
```
Comment (swordbeta) [inline Lib/Dropbox/DropboxAuthManager.php]: Remove double semicolon

File: Lib/Dropbox/DropboxAuthManager.php [hunk 4]
```diff
@@ -0,0 +1,91 @@
...
+	 * @param $userId
+	 *
+	 * @return DropboxAuthContainer
+	 */
+	public function getAuthContainer($userId) {
+		/** @var MediaPlatformUser $MediaPlatformUser */
+		$MediaPlatformUser = ClassRegistry::init('AuthManager.MediaPlatformUser');
+		$dropboxAuthContainer = new DropboxAuthContainer();
+		$oauthTokens = $MediaPlatformUser->getOauthTokens($userId);
+		$dropboxAuthContainer->dropbox = new Dropbox\Dropbox($oauthTokens['OauthToken']['access_token']);
```
Comment (swordbeta) [inline Lib/Dropbox/DropboxAuthManager.php]: Add this dependency to `composer.json`

**Comment by ronoudgenoeg** [comment]
@swordbeta Done!

**Pull request merged**